web/source/kmwdevice.ts

```diff
@@ -70,7 +70,54 @@ namespace com.keyman {
         } else if(agent.indexOf('Linux') >= 0) {
           this.OS='Linux';
         } else if(agent.indexOf('Macintosh') >= 0) {
-          this.OS='MacOSX';
+          // Starting with 13.1, "Macintosh" can reflect iPads (by default) or iPhones 
+          // (by user setting); a new "Request Desktop Website" setting for Safari will
+          // change the user agent string to match a desktop Mac.
+          let regex = /Intel Mac OS X (10(?:_\d+)+)/i;
+          let results = regex.exec(agent);
+          
+          // Match result:  a version string with components separated by underscores.
+          if(results.length > 1 && results[1]) {
+            // Converts version string to array of numbers in major.minor.build order.
+            // Example:  10_14_6 => [10, 14, 6]
+            let versionComponentStrings = results[1].split('_');
```

Comment:
What if you tweak the Version() constructor (version.ts l.21) to use a delimiter as a second parameter?

Then you could do `let version = new utils.Version(results[1], '_');`

Reply:
I suppose I _could_ do that, but then the constructor's signature starts getting complex than it already is; TypeScript doesn't allow constructor overloading.

Then again, I just had a different idea that's simpler... why not just `.replace('_', '.')`?